src/Symfony/Component/HttpKernel/Event/ControllerEvent.php

```diff
@@ -43,6 +44,30 @@ public function getController(): callable
 
     public function setController(callable $controller): void
     {
+        if (isset($this->controller) && $controller !== $this->controller) {
+            $this->getRequest()->attributes->remove('_controller_reflectors');
+        }
+
         $this->controller = $controller;
     }
+
+    public function getAttributes(): array
+    {
+        if (isset($this->attributes) || ![$class, $action] = $this->getRequest()->attributes->get('_controller_reflectors')) {
+            return $this->attributes ??= [];
+        }
+
+        foreach (array_merge($class?->getAttributes() ?? [], $action->getAttributes()) as $attribute) {
+            if (class_exists($attribute->getName())) {
+                $this->attributes[$attribute->getName()][] = $attribute->newInstance();
+            }
+        }
+
+        return $this->attributes;
```

Comment:
to me, this is missing an assignment before the loop, to ensure that it is an array even if the loop does not do any assignment